wordpress-mobile/WordPress-FluxC-Android
# Woo: Delete WCOrderSummaryModels for site in db when fetchFirstPage() is called

This PR fixes #1272 by deleting all `WCOrderSummaryModel`s belonging to a site from the db when the first page is being fetched. This PR also adds some unit tests for `WCOrderSummaryModel`-related methods in `OrderSqlUtils`. I even included a test to verify the foreign key relationship between `SiteModel` and `WCOrderSummaryModel`.

## To Test
- Run the example app and navigate to the Order List page via **Woo -> Orders -> Fetch Order List**
- Do a pull to refresh and verify all records populate as expected.
- You can also put a breakpoint [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/2a8906c3de49d71455555255b855f80b0016fd8a/plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCOrderStore.kt#L519) to make sure the table is only purged when the list is first being fetched (ie: `loadMore` is false).
- Run the tests in `OrderSqlUtilsTest`
